### PR TITLE
Fix/#199 Propagate changes from `taipy-gui` and `taipy-enterprise`

### PR DESCRIPTION
**Pull request by trgiangdo**
https://app.zenhub.com/workspaces/engineering-team-6315bfd19a5b920013dabbaf/issues/avaiga/taipy-doc/199

I've also add a commit to fix `os.path.dirname` return empty string on Ubuntu machine. [Link to the solution and explanation.](https://stackoverflow.com/questions/7783308/os-path-dirname-file-returns-empty)

### CODE REVIEW

File: tools/setup_generation.py [hunk 1]
```diff
@@ -40,7 +40,7 @@
 PACKAGE_GROUP = [ "taipy.config", "taipy.core", "taipy.gui", "taipy.rest", "taipy.auth", "taipy.enterprise" ]
 
 # Assuming that this script is located in <taipy-doc>/tools
-tools_dir = os.path.dirname(__file__).replace("\\", "/")
+tools_dir = os.path.dirname(os.path.abspath(__file__)).replace("\\", "/")
```
Comment (FabienLelaquais) [inline tools/setup_generation.py:43]: What was the problem with that?

File: Pipfile [hunk 2]
```diff
@@ -9,12 +9,13 @@ Flask-RESTful = "==0.3.9"
 apispec = {extras = ["yaml"], version = "==5.0"}
 apispec-webframeworks = "==0.5.2"
 bcrypt = "==3.2.2"
-flask = "==2.1"
+flask = "==2.2.2"
 flask-cors = "==3.0.10"
 flask-marshmallow = "==0.14"
 flask-socketio = "==5.1.1"
```
Comment (FabienLelaquais) [inline Pipfile]: Actually all this section should be changed as well:
```
flask = "==2.1"
flask-cors = "==3.0.10"
flask-marshmallow = "==0.14"
flask-socketio = "==5.1.1"
flask-talisman = "==1.0"
```
=>
```
flask = "==2.2.2"
flask-cors = "==3.0.10"
flask-marshmallow = "==0.14"
flask-socketio = "==5.3.0"
gevent = "==21.12.0"
gevent-websocket = "==0.10.1"
kthread = "==0.2.3"
```

**Review by FabienLelaquais — APPROVED** [review]
I have slightly different entries in Pipenv, as you'll see.
But well if it works, it works...
Thx!

File: tools/setup_generation.py (hunk 1 above)
Comment (trgiangdo) [reply tools/setup_generation.py:43]: On my Ubuntu WSL, the `os.path.dirname(__file__)` actually return an empty string, which is weird. I also did some searching and found others with similar issue on other Linux distro.

File: Pipfile (hunk 2 above)
Comment (trgiangdo) [inline Pipfile:15]:
```suggestion
flask-socketio = "==5.3.0"
gevent = "==21.12.0"
gevent-websocket = "==0.10.1"
```
This should be ok, I suppose.
Does it necessary to move `gevent` and `gevent-websocket` below `flask-talisman` line?

Comment (trgiangdo) [reply Pipfile]:
```suggestion
flask-socketio = "==5.3.0"
gevent = "==21.12.0"
gevent-websocket = "==0.10.1"
```
This should be ok, I suppose.
Does it necessary to move `gevent` and `gevent-websocket` below `flask-talisman` line?

Comment (FabienLelaquais) [reply Pipfile]: The order is important only for us human beings: easier to locate an entry

**Review by jrobinAV — APPROVED** [review]
Looks good to me.
Thx

**Pull request merged**